actionview/lib/action_view/template/resolver.rb

```diff
@@ -136,8 +129,12 @@ def find_all_anywhere(name, prefix, partial=false, details={}, key=nil, locals=[
       end
     end
 
-    def find_all_with_query(query) # :nodoc:
-      @cache.cache_query(query) { find_template_paths(File.join(@path, query)) }
+    def find_all_unescaped(name, prefix, partial = false, details = {}, key = nil, locals = []) #:nodoc:
+      cached(key, [name, prefix, partial], details, locals) do
```

Comment:
Looks like we gotta be careful here, security wise. What happens if we cache a template with the path unescaped and then query it by some other means and get the template back?